src/storage-types/src/sources.rs

```diff
@@ -1482,9 +1490,15 @@ impl<C: ConnectionAccess> SourceConnection for KafkaSourceConnection<C> {
             .iter()
             .map(|(name, kind)| {
                 let typ = match kind {
-                    KafkaMetadataKind::Partition => ScalarType::Int32,
-                    KafkaMetadataKind::Offset => ScalarType::UInt64,
-                    KafkaMetadataKind::Timestamp => ScalarType::Timestamp { precision: None },
+                    KafkaMetadataKind::Partition => ScalarType::Int32.nullable(false),
+                    KafkaMetadataKind::Offset => ScalarType::UInt64.nullable(false),
+                    KafkaMetadataKind::Timestamp => {
+                        ScalarType::Timestamp { precision: None }.nullable(false)
+                    }
+                    KafkaMetadataKind::Header { use_bytes, .. } => use_bytes
+                        .then_some(ScalarType::Bytes)
+                        .unwrap_or(ScalarType::String)
```

Comment:
Going through `then_some` is a bit unnecessary and hard to read here since you're jumping out of the `Option` world immediately. I would write it like this:

```rust
KafkaMetadataKind::Header { use_bytes: true, .. } => ScalarType::Bytes.nullable(true),
KafkaMetadataKind::Header { use_bytes: false, .. } => ScalarType::String.nullable(true),
```